### ISSUE TITLE
minio uses wrong response_type and has duplicated items in scope

### ISSUE THREAD

**Issue by UchihaYuki** [minio/console#1641]

**Comment by harshavardhana** [comment]
The issue in the usage of AuthCodeURL()

**Comment by harshavardhana** [comment]
> * set `response_types_supported: ["id_token"]` and `scopes_supported: ["openid"]` in `https://my-iam.com/.well-known/openid-configuration`

You shouldn't be restricting your IDP in this manner we require "code" as well as "id_token" 

> ```
> https://my-iam.com/authorization?client_id=cluster&redirect_uri=https%3A%2F%2Fmy-s3.com&**response_type=code**&**scope=openid+openid**&state=TVlOU0lXRlNLMVlWNTNSMjFKWTVaRlRQVDpzOTRxak5iNGgyb3gySjJaN0ozdG5NdEhGZ3ErWFM4MElYdW9kbUx3Wjc0PQ%3D%3D
> ```

You should fix your IDP configuration.

**Comment by UchihaYuki** [comment]
After setting my `/.well-known/openid-configuration` to:
```
response_types_supported: [
    "code",
    "token",
    "id_token",
    "code token",
    "code id_token",
    "token id_token",
    "code token id_token",
    "none",
],
scopes_supported: ["openid", "email", "profile"],
```
It will choose `code` over `code id_token`, although I can return id_token as access_token to get around it in my case, which might be a problem for others.

Another problem is, it uses two openid entries in request URL:
 
/authorization?client_id=cluster&redirect_uri=https%3A%2F%2Fmy-iam.com%3A8443%2Foauth_callback&response_type=code&scope=**openid+openid**&state=WjJTV0ZGQ1BaSUtUQkpLQ00wT1ZBV0tWVDpWd0pZdE12ZWdSdUI5K3RUM0pmTk9hY1Job2FLbENueVU4RWtSYWxwOUFnPQ%3D%3D

**Comment by harshavardhana** [comment]
None of those are problems with real IDP such as auth0. I don't know what you are doing here. 

It looks like some artisanal OpenID implementation.

**Comment by harshavardhana** [comment]
> I guess this is the source of the repetation.

Yeah, that is not a problem - look at other OpenID implementations on how to handle this. Multiple scopes are simply just one scope.
